go/worker/registration/worker.go

```diff
@@ -899,6 +899,16 @@ func (w *Worker) registerNode(epoch beacon.EpochTime, hook RegisterNodeHook) err
 	}
 
 	if err := hook(&nodeDesc); err != nil {
+		w.Lock()
+		w.status.Successful = false
```

Comment:
would it be possible to do this update outside of registerNode so that we don't need to replicate this for every error return